crates/external_services/src/grpc_client.rs

```diff
@@ -76,3 +78,34 @@ impl GrpcClientSettings {
         })
     }
 }
+
+/// Contains grpc headers
+#[derive(Debug)]
+pub struct GrpcHeaders {
+    /// Tenant id
+    pub tenant_id: String,
+    /// Request id
+    pub request_id: Option<String>,
+}
+
+impl GrpcHeaders {
+    /// Method to add necessary headers to the tonic Request
+    pub fn add_headers_to_grpc_request<T>(&self, request: &mut tonic::Request<T>) {
+        self
+            .tenant_id
+            .parse()
+            .map(|tenant_id| request.metadata_mut().append("x-tenant-id", tenant_id))
```

Comment:
Nit: Can use constants instead of string literals. There's one for `x-tenant-id` in the `common_utils` crate.